### PR TITLE
Add org.jacktrip.JackTrip

### PR DESCRIPTION
**Pull request by ntonnaett**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

~~I'm using `--socket=system-bus` because I've read that this is necessary so that JackTrip can give the network thread realtime priority.~~

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.jacktrip.JackTrip.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+app-id: org.jacktrip.JackTrip
+runtime: org.kde.Platform
+runtime-version: '5.15'
```
Comment (hfiguiere) [inline org.jacktrip.JackTrip.yml:3]:
```suggestion
runtime-version: '5.15-21.08'
```

File: org.jacktrip.JackTrip.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  - --share=ipc
+  - --socket=x11
+  # Wayland access
+  # - --socket=wayland
+  # Needs network access
+  - --share=network
+  # Pipewire/Jack
+  - --filesystem=xdg-run/pipewire-0
+  # For setting realtime priority for network thread?
+  - --socket=system-bus
```
Comment (hfiguiere) [inline org.jacktrip.JackTrip.yml:17]:
```suggestion
  - --system-talk-name=org.freedesktop.RealtimeKit1
```

But it doesn't work (ie it doesn't set the real time priority it's a sandboxing limitation)

File: org.jacktrip.JackTrip.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    cleanup: [ "*" ]
+    build-commands:
+    - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "jinja2" --no-build-isolation
+    sources:
+    - sha256: 594c67807fb16238b30c44bdf74f36c02cdf22d1c8cda91ef8a0ed8dabf5620a
+      type: file
+      url: https://files.pythonhosted.org/packages/bf/10/ff66fea6d1788c458663a84d88787bae15d45daa16f6b3ef33322a51fc7e/MarkupSafe-2.0.1.tar.gz
+    - sha256: 077ce6014f7b40d03b47d1f1ca4b0fc8328a692bd284016f806ed0eaca390ad8
+      type: file
+      url: https://files.pythonhosted.org/packages/20/9a/e5d9ec41927401e41aea8af6d16e78b5e612bca4699d417f646a9610a076/Jinja2-3.0.3-py3-none-any.whl
```
Comment (hfiguiere) [inline org.jacktrip.JackTrip.yml:43]:
```suggestion
    - type: file
      sha256: 594c67807fb16238b30c44bdf74f36c02cdf22d1c8cda91ef8a0ed8dabf5620a
      url: https://files.pythonhosted.org/packages/bf/10/ff66fea6d1788c458663a84d88787bae15d45daa16f6b3ef33322a51fc7e/MarkupSafe-2.0.1.tar.gz
    - type: file
      sha256: 077ce6014f7b40d03b47d1f1ca4b0fc8328a692bd284016f806ed0eaca390ad8
      url: https://files.pythonhosted.org/packages/20/9a/e5d9ec41927401e41aea8af6d16e78b5e612bca4699d417f646a9610a076/Jinja2-3.0.3-py3-none-any.whl
```

File: org.jacktrip.JackTrip.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  # Wayland access
+  # - --socket=wayland
+  # Needs network access
+  - --share=network
+  # Pipewire/Jack
+  - --filesystem=xdg-run/pipewire-0
+  # For setting realtime priority for network thread?
+  - --socket=system-bus
+cleanup:
+  - /lib/python3.8
```
Comment (hfiguiere) [inline org.jacktrip.JackTrip.yml:19]:
```suggestion
  - /lib/python3.9
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.jacktrip.JackTrip.yml (hunk 2 above)
Comment (ntonnaett) [reply org.jacktrip.JackTrip.yml:17]: AFAIK RealtimeKit doesn't even work for JACK or Pipewire.

File: org.jacktrip.JackTrip.yml (hunk 1 above)
Comment (ntonnaett) [reply org.jacktrip.JackTrip.yml:3]: Is there any documentation about the runtime versioning?

File: org.jacktrip.JackTrip.yml (hunk 2 above)
Comment (hfiguiere) [reply org.jacktrip.JackTrip.yml:17]: if it's a different dbus-name then use the right one. But `--socket=system-bus` is a no no in general.

**Comment by hfiguiere** [comment]
bot, build org.jacktrip.JackTrip

File: org.jacktrip.JackTrip.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  - /share/man
+modules:
+  - shared-modules/linux-audio/jack2.json
```
Comment (hfiguiere) [inline org.jacktrip.JackTrip.yml:22]: you need to had the git submodule for this to work.

File: org.jacktrip.JackTrip.yml (hunk 2 above)
Comment (ntonnaett) [reply org.jacktrip.JackTrip.yml]: I removed it entirely.

File: org.jacktrip.JackTrip.yml (hunk 5 above)
Comment (ntonnaett) [reply org.jacktrip.JackTrip.yml:20]: I was already wondering why this doesn't happen on my machine. One git reset too many.

**Comment by hfiguiere** [comment]
bot, build org.jacktrip.JackTrip

**Comment by ntonnaett** [comment]
There are graphics driver warnings on my system. Do I need extra permissions for QT?

**Comment by ntonnaett** [comment]
Fixed the libGL error messages.

**Comment by hfiguiere** [comment]
Yeah `--device=dri` is usually wanted too.

**Comment by hfiguiere** [comment]
bot, build org.jacktrip.JackTrip

**Pull request closed without merging**